src/middleware/inject_params.rs

```diff
@@ -60,7 +84,14 @@ impl Middleware<CallRequest, Result<JsonValue, Error>> for InjectParamsMiddlewar
                 let to_inject = self.get_parameter().await;
                 tracing::debug!("Injected param {} to method {}", &to_inject, request.method);
                 while request.params.len() < idx {
-                    request.params.push(JsonValue::Null);
+                    let current = request.params.len();
+                    if self.params[current].is_optional == Some(true) {
+                        request.params.push(JsonValue::Null);
+                    } else {
+                        return Err(Error::Call(CallError::InvalidParams(anyhow::Error::msg(
+                            "non-optional param",
```

Comment:
can you check what's the error message from Substrate RPC and try to use the same one

Reply:
there is no error msg from substrate RPC here no?

Reply:
I mean if you send a rpc call to substrate directly with missing required parameter, what will be the error message? just trying to make it behave the same